### PR TITLE
Adding STEVAL-PROTEUS1 device model

### PR DESCRIPTION
**Pull request by dmelpi**
### Thank you for contributing to the Azure IoT Plug and Play Models repository

This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

The repository pipeline will ensure minimum validation of incoming DTDL models.  

- PR validation steps are described in the tools [Wiki](https://github.com/Azure/iot-plugandplay-models-tools/wiki/Validation-Pipeline#pr-validation-checks).
- The same validation can be run locally via the [dmr-client](https://github.com/Azure/iot-plugandplay-models-tools/tree/dev/clients/dotnet/Microsoft.IoT.ModelsRepository.CommandLine) command line tool.

## Requested info template for model(s) submission

When submitting models to the repository we ask that you provide as much of the following meta information around your models and related devices as possible. This info will be used to improve Plug and Play.

:star2: Please replace the markdown comment examples with your own values.

### CODE REVIEW

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 1]
```diff
@@ -0,0 +1,301 @@
+{
+  "@context": "dtmi:dtdl:context;2",
+  "@id": "dtmi:stmicroelectronics:steval_proteus1:standard_if;1",
+  "@type": "Interface",
+  "description": {
+    "en": "Reports temperature, accelerometer, gyroscope, predictive speed status, predictive acceleration peak status, predictive frequency domain status, battery status"
+  },
+  "displayName": {
+    "en": "Standard interface for PROTEUS"
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:9]: "interface" should be capitalized.

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+  },
+  "contents": [
+    {
+      "@type": "Telemetry",
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:13]: could use co-type "Unitless" and unit "percent"

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+      "comment": "Acceleration X/Y/Z components [mg]",
+      "displayName": {
+        "en": "Acc Value [mg]"
+      },
+      "name": "accelerometer",
+      "schema": {
+        "@type": "Object",
+        "description": {
+          "en": "Single acceleration sample. I.e.: {a_x, a_y, a_z}."
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:44]: Per [DTDL Naming Guidelines](https://github.com/Azure/iot-plugandplay-models-tools/wiki/DTDL-Naming-Guidelines), "acceleration" and "sample" should be capitalized.

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+        },
+        "displayName": {
+          "en": "Single acceleration sample."
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:47]: Per [DTDL Naming Guidelines](https://github.com/Azure/iot-plugandplay-models-tools/wiki/DTDL-Naming-Guidelines), "acceleration" and "sample" should be capitalized.  Also, generally better not to have a period at the end of displayName, only at the end of description.

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+        "en": "Gyro Value [dps]"
+      },
+      "name": "gyroscope",
+      "schema": {
+        "@type": "Object",
+        "description": {
+          "en": "Single angular velocity sample. I.e.: {g_x, g_y, g_z}."
+        },
+        "displayName": {
+          "en": "Single angular velocity sample."
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:78]: Capitalize all words and remove terminal period.

File: dtmi/stmicroelectronics/steval_proteus1/standard_fw-1.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  "@context": "dtmi:dtdl:context;2",
+  "@id": "dtmi:stmicroelectronics:steval_proteus1:standard_fw;1",
+  "@type": "Interface",
+  "displayName": "STEVAL-PROTEUS1",
+  "description": "2.4GHz industrial sensor node for condition monitoring.",
+  "comment": "from proteus-poc iotc draftv2 (no deviceinfo component)",
+  "contents": [
+    {
+      "@type": "Component",
+      "name": "stdcomp",
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_fw-1.json:11]: camelCase is preferred for "name" values, so perhaps this should be "stdComp" ?

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+        ]
+      }
+    },
+    {
+      "@type": "Telemetry",
+      "comment": "Acceleration Peak Status X/Y/Z components [m/sˆ2]",
+      "displayName": {
+        "en": "Acc Peak Status [m/sˆ2]"
+      },
+      "name": "predictiveaccelerationstatus",
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:102]: camelCase name: predictiveAccelerationStatus

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+      "schema": {
+        "@type": "Object",
+        "description": {
+          "en": "Sample format I.e.: {statusacc_x, statusacc_y, statusacc_z, accpeak_x, accpeak_y, accpeak_z}."
+        },
+        "displayName": {
+          "en": "Single Predictive Acceleration Peak Status sample."
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:109]: remove terminal period?

Comment (jrdouceur) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:109]: ... and capitalize "sample"

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+        },
+        "fields": [
+          {
+            "displayName": {
+              "en": "[AccPeak] Status Acceleration X:"
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:114]: Remove ':' from end of displayName value, here and below

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 10]
```diff
@@ -0,0 +1,301 @@
...
+              "en": "[AccPeak] Status Acceleration Z:"
+            },
+            "name": "statusacc_z",
+            "schema": "double"
+          },
+          {
+            "displayName": {
+              "en": "Acceleration Peak X:"
+            },
+            "name": "accpeak_x",
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:137]: Perhaps capital P here and below, as in "accPeak"

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+            "schema": "double"
+          }
+        ]
+      }
+    },
+    {
+      "@type": "Telemetry",
+      "comment": "Speed Status X/Y/Z components [mm/s]",
+      "displayName": {
+        "en": "Speed Status  [m/sˆ2]"
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:161]: spurious double space

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+      },
+      "name": "predictivespeedstatus",
+      "schema": {
+        "@type": "Object",
+        "description": {
+          "en": "Sample format I.e.: {statusacc_x, statusacc_y, statusacc_z, rmsspeed_x, rmsspeed_y, rmsspeed_z}."
+        },
+        "displayName": {
+          "en": "Single Predictive Speed Status sample."
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:170]: capitalize "sample" and remove terminal period

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 13]
```diff
@@ -0,0 +1,301 @@
...
+        },
+        "fields": [
+          {
+            "displayName": {
+              "en": "[Speed] Status Acceleration X:"
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:175]: remove terminal colon, here and below

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 14]
```diff
@@ -0,0 +1,301 @@
...
+            },
+            "name": "statusacc_x",
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:177]: perhaps capital A, as in "statusAcc" here and below

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 15]
```diff
@@ -0,0 +1,301 @@
...
+              "en": "[Speed] Status Acceleration Z:"
+            },
+            "name": "statusacc_z",
+            "schema": "double"
+          },
+          {
+            "displayName": {
+              "en": "RMS SPEED X:"
+            },
+            "name": "rmsspeed_x",
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:198]: perhaps capital S, here and below: "rmsSpeed"

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 16]
```diff
@@ -0,0 +1,301 @@
...
+        ]
+      }
+    },
+    {
+      "@type": "Telemetry",
+      "comment": "Speed Status X/Y/Z components [mm/s]",
+      "displayName": {
+        "en": "Speed Status  [m/sˆ2]"
+      },
+      "name": "predictivespeedstatus",
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:163]: camelCase name: predictiveSpeedStatus

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 17]
```diff
@@ -0,0 +1,301 @@
...
+        ]
+      }
+    },
+    {
+      "@type": "Telemetry",
+      "comment": "Frequency Domain Status X/Y/Z components [mm/s]",
+      "displayName": {
+        "en": "Frequency Domain Status  [m/sˆ2]"
+      },
+      "name": "predictivefrequencydomainstatus",
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:224]: camelCase name: predictiveFrequencyDomainStatus

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 18]
```diff
@@ -0,0 +1,301 @@
...
+        "description": {
+          "en": "Sample format I.e.: {statusacc_x, statusacc_y, statusacc_z, freq_x, freq_y, freq_z, maxampli_x, maxampli_y, maxampli_z}."
+        },
+        "displayName": {
+          "en": "Single Predictive Frequency Domain sample."
+        },
+        "fields": [
+          {
+            "displayName": {
+              "en": "[Frequency Domain] Status Acceleration X:"
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:236]: remove terminal colon, here and below

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 19]
```diff
@@ -0,0 +1,301 @@
...
+              "en": "[Frequency Domain] Status Acceleration Y:"
+            },
+            "name": "statusacc_y",
+            "schema": "double"
+          },
+          {
+            "displayName": {
+              "en": "[Frequency Domain] Status Acceleration Z:"
+            },
+            "name": "statusacc_z",
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:252]: perhps capital S: "statusAcc"

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 20]
```diff
@@ -0,0 +1,301 @@
...
+              "en": "Frequency Domain Z:"
+            },
+            "name": "freq_z",
+            "schema": "double"
+          },
+          {
+            "displayName": {
+              "en": "Max Amplitude X:"
+            },
+            "name": "maxampli_x",
```
Comment (jrdouceur) [inline dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:280]: perhaps capital A, here and below: "maxAmpli"

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 21]
```diff
@@ -0,0 +1,301 @@
+{
+  "@context": "dtmi:dtdl:context;2",
+  "@id": "dtmi:stmicroelectronics:steval_proteus1:standard_if;1",
+  "@type": "Interface",
+  "description": {
+    "en": "Reports temperature, accelerometer, gyroscope, predictive speed status, predictive acceleration peak status, predictive frequency domain status, battery status"
```
[the remark this answers is not in the document]
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:6]: ok, will fix that

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 1 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:9]: ok

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 4 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:47]: ok

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 21 above)
[the remark this answers is not in the document]
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:6]: what is the preferred way forward: should I fix all changes and submit a new pull request?

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 2 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:13]: tried but fails validation

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 5 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: fixed

File: dtmi/stmicroelectronics/steval_proteus1/standard_fw-1.json (hunk 6 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_fw-1.json:11]: changed to std_comp, according to our current software conventions

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 8 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: done

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 9 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: done

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 11 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: done

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 13 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: done

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 12 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: done

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 22]
```diff
@@ -0,0 +1,301 @@
...
+        "en": "Frequency Domain Status  [m/sˆ2]"
+      },
+      "name": "predictivefrequencydomainstatus",
+      "schema": {
+        "@type": "Object",
+        "description": {
+          "en": "Sample format I.e.: {statusacc_x, statusacc_y, statusacc_z, freq_x, freq_y, freq_z, maxampli_x, maxampli_y, maxampli_z}."
+        },
+        "displayName": {
+          "en": "Single Predictive Frequency Domain sample."
```
[the remark this answers is not in the document]
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: done

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 18 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: done

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 7 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: done in [335ab63](https://github.com/Azure/iot-plugandplay-models/pull/449/commits/335ab63428b102711a254df3e3b7f64320e9601a)

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json [hunk 23]
```diff
@@ -0,0 +1,301 @@
...
+        },
+        "displayName": {
+          "en": "Single Predictive Acceleration Peak Status sample."
+        },
+        "fields": [
+          {
+            "displayName": {
+              "en": "[AccPeak] Status Acceleration X:"
+            },
+            "name": "statusacc_x",
```
[the remark this answers is not in the document]
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: [335ab63](https://github.com/Azure/iot-plugandplay-models/pull/449/commits/335ab63428b102711a254df3e3b7f64320e9601a)

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 10 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: [335ab63](https://github.com/Azure/iot-plugandplay-models/pull/449/commits/335ab63428b102711a254df3e3b7f64320e9601a)

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 14 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: [335ab63](https://github.com/Azure/iot-plugandplay-models/pull/449/commits/335ab63428b102711a254df3e3b7f64320e9601a)

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 15 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: [335ab63](https://github.com/Azure/iot-plugandplay-models/pull/449/commits/335ab63428b102711a254df3e3b7f64320e9601a)

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 16 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: [335ab63](https://github.com/Azure/iot-plugandplay-models/pull/449/commits/335ab63428b102711a254df3e3b7f64320e9601a)

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 17 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: [335ab63](https://github.com/Azure/iot-plugandplay-models/pull/449/commits/335ab63428b102711a254df3e3b7f64320e9601a)

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 19 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: [335ab63](https://github.com/Azure/iot-plugandplay-models/pull/449/commits/335ab63428b102711a254df3e3b7f64320e9601a)

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 20 above)
Comment (dmelpi) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json]: [335ab63](https://github.com/Azure/iot-plugandplay-models/pull/449/commits/335ab63428b102711a254df3e3b7f64320e9601a)

File: dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json (hunk 2 above)
Comment (jrdouceur) [reply dtmi/stmicroelectronics/steval_proteus1/standard_if-1.json:13]: My bad.  DTDL does not have a generic Unitless semantic type, only specific types that are unitless, such as RelativeHumidity.  It also does not have a BatteryLevel or similarly appropriate type.

File: dtmi/stmicroelectronics/steval_proteus1/standard_fw-1.json (hunk 6 above)
Comment (jrdouceur) [reply dtmi/stmicroelectronics/steval_proteus1/standard_fw-1.json]: Fine by me.

**Pull request merged**